qpc/insights/utils.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2018 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 3 (GPLv3). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv3
+# along with this software; if not, see
+# https://www.gnu.org/licenses/gpl-3.0.txt.
+#
+"""Utilities for the insights module."""
+
+from __future__ import print_function
+
+# pylint: disable=no-name-in-module,import-error
+from distutils.version import LooseVersion
+
+
+class InsightsCommands():
+    """Creates insights-client commands for qpc client."""
+
+    def __init__(self, test=False):
+        """Set class variables.
+
+        Args:
+            test (bool): used to test insights commands locally.
+        """
+        self.test = test
+        # mime type for QPC report data for uploading to insights
+        self.content_type = 'application/vnd.redhat.qpc.test+tgz'
```

Comment:
Based on this documentation, 
```
^application/vnd\.redhat\.([a-z0-9-]+)\.([a-z0-9-]+)\+(tgz|zip)$, where the first matching group is a unique name for an Insights Platform application, the second matching group identifies the payload type for that application, and the third matching group identifies whether the payload is a gzipped tarball or zip file. Any other Content-Type header values will be met with a 415 Unsupported Media Type response. These supported types should be discoverable with a GET request.
```

Should the content type be `application/vnd.redhat.qpc.deployments.tgz`?

Reply:
I was not certain if we had to have the insights team build us an endpoint or not? Honestly, I am not exactly sure how they use content type/mime type.